optuna/samplers/_nsgaiii.py

```diff
@@ -118,6 +118,12 @@ def __init__(
                 " The interface can change in the future.",
                 ExperimentalWarning,
             )
+        if after_trial_strategy is not None:
+            warnings.warn(
+                "The after_trial_strategy option is an experimental feature."
+                " The interface can change in the future.",
+                ExperimentalWarning,
+            )
```

Comment:
I think this warning is not necessary because all of `NSGAIIISampler` is experimental.